osu.Game/Beatmaps/BeatmapManager_WorkingBeatmap.cs

```diff
@@ -21,16 +19,12 @@ public partial class BeatmapManager
         [ExcludeFromDynamicCompile]
         private class BeatmapManagerWorkingBeatmap : WorkingBeatmap
         {
-            private readonly IResourceStore<byte[]> store;
-            private readonly TextureStore textureStore;
-            private readonly ITrackStore trackStore;
+            private readonly IBeatmapResourceProvider resources;
 
-            public BeatmapManagerWorkingBeatmap(IResourceStore<byte[]> store, TextureStore textureStore, ITrackStore trackStore, BeatmapInfo beatmapInfo, AudioManager audioManager)
-                : base(beatmapInfo, audioManager)
+            public BeatmapManagerWorkingBeatmap(BeatmapInfo beatmapInfo, IBeatmapResourceProvider resources)
+                : base(beatmapInfo, resources?.AudioManager)
```

Comment:
Sort of confused by the null-check on `resources` there, especially seeing that the rest of this class assumes the converse, and the only caller of this ctor passes `this`. Would remove this null-check (and maybe annotate param as `[NonNull]`).